modules/gdscript/gdscript.cpp

```diff
@@ -1381,8 +1381,17 @@ void GDScript::_save_orphaned_subclasses() {
 	}
 
 	// clear subclasses to allow unused subclasses to be deleted
+	for (KeyValue<StringName, Ref<GDScript>> &E : subclasses) {
+		p_clear_data->scripts.insert(E.value);
+	}
 	subclasses.clear();
 	// subclasses are also held by constants, clear those as well
+	for (KeyValue<StringName, Variant> &E : constants) {
+		GDScript *gdscr = _get_gdscript_from_variant(E.value);
```

Comment:
Do you need to consider situations like `const EnumNewName = SomeOtherClass.NamedEnum`? Does this count as some sort of weak reference?

Reply:
You cannot have enums without a `GDScript` instance.

Reply:
I trust what you say! It just occurred to me as someone who isn't knowledgeable about this code :)

So basically, `SomeOtherClass` had to show up somewhere in the code, and therefore has already been taken into account somewhere else.

What happens with
```GDScript
class_name A
  const B = preload("b.gd")
```

I take it it is also accounted for as dependency by this code? Or is it taken to be a subclass?


Let me know if this isn't useful - this is me taking shots in the dark trying to make sense of the code :)

Reply:
> What happens with
> 
> ```gdscript
> class_name A
>   const B = preload("b.gd")
> ```
> 
> I take it it is also accounted for as dependency by this code? Or is it taken to be a subclass?

From that example, `B` is a dependency of `A`

Reply:
> Or is it taken to be a subclass?

A subclass is a class within a gdscript file.

```
class_name ThisIsNotASubclass

class ThisIsASubclass:
    pass
```

Reply:
Hahah, the naming of all this is messing with me :) I would call that an inner class, whereas a subclass would be `class B extends A`!

Sounds like you've gotten all this figured out though, so I'm happy to see this merged! The changes make sense!